cy_ioctl.go

```diff
@@ -154,27 +154,42 @@ func (d *Device) doIoctl(kind uint32, iface whd.IoctlInterface, cmd whd.SDPCMCom
 		return errors.New("invalid ioctl kind")
 	}
 
+	d.hw.Lock()
```

Comment:
Please add these lock checks in the same place as the original CY_43 driver has them. I believe they appear as CYW43_THREAD_ENTER for `Lock`, CYW43_THREAD_EXIT for `Unlock`, and CYW43_THREAD_LOCK_CHECK for `if !TryLock {return} `.

I'd also suggest we add `lock`, `unlock` methods on `Device` that wrap these so we can add debug printing to when the device locks/unlocks.

Reply:
done, double check please